shared/src/api.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * Konveyor Core Extension API
+ * Types for inter-extension communication between core and language extensions
+ */
+
+/**
+ * Initialization configuration for a provider instance
+ */
+export interface ProviderInitConfig {
+  /** Workspace location to analyze */
+  location: string;
+
+  /** Analysis mode (e.g., "source-only", "full-with-deps") */
+  analysisMode: string;
+
+  /** Named pipe/socket path for LSP proxy communication (JSON-RPC) */
+  pipeName: string;
+}
+
+/**
+ * Configuration for a language provider that will be passed to kai-analyzer-rpc
+ */
+export interface ProviderConfig {
+  /** Provider name (e.g., "java", "python") */
+  name: string;
+
+  /** Address/path where kai-analyzer-rpc can connect to the provider (GRPC socket) */
+  address: string;
+
+  /** Whether to use sockets/named pipes for communication (true for UDS/Windows named pipes) */
+  useSockets?: boolean;
+
+  /** Initialization configuration for the provider */
+  initConfig?: ProviderInitConfig[];
+
+  /** Number of context lines to include around incidents */
+  contextLines?: number;
+}
+
+/**
+ * Metadata about a provider's supported migration bundles
+ */
+export interface BundleMetadata {
```

Comment:
nit: BundleMetadata is unclear and confusing because of existence of java bundle

Reply:
I'm dropping this for now. If we have a need for it we can update the API to include it and name it then.